src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,373 @@
+    <Setter Property="FocusVisualMargin" Value="-3" />-->
+    <Setter Property="KeyboardNavigation.IsTabStop" Value="True" />
+    <Setter Property="Padding" Value="{DynamicResource ButtonPadding}" />
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate>
+          <!-- A Border was added here since Avalonia does not support Background/Border/CornerRadius on Grid directly -->
+          <!-- ClipToBounds added as a temporary work-around for: https://github.com/AvaloniaUI/Avalonia/issues/2105 -->
+          <Border Background="Transparent"
+                  CornerRadius="{TemplateBinding CornerRadius}"
+                  ClipToBounds="True">
+            <Border.Styles>
+              <!-- Override the style of the inner buttons so that they don't affect background/foreground/border colors -->
+              <Style Selector="Button">
+                <Setter Property="Foreground" Value="{DynamicResource SplitButtonForeground}" />
+                <Setter Property="BorderBrush" Value="Transparent" />
+                <Setter Property="BorderThickness" Value="{DynamicResource SplitButtonBorderThemeThickness}" />
+                <Setter Property="HorizontalAlignment" Value="Left" />
+                <Setter Property="VerticalAlignment" Value="Center" />
+                <!--<Setter Property="FontFamily" Value="{ThemeResource ContentControlThemeFontFamily}" />
+                <Setter Property="FontWeight" Value="Normal" />
+                <Setter Property="FontSize" Value="{ThemeResource ControlContentThemeFontSize}" />
+                <Setter Property="UseSystemFocusVisuals" Value="{StaticResource UseSystemFocusVisuals}" />
+                <Setter Property="FocusVisualMargin" Value="-3" />-->
+                <Setter Property="Template">
+                  <ControlTemplate>
+                    <!-- Switched from Grid to Border to optimize and support Foreground/Background properties -->
+                    <Border x:Name="RootBorder"
+                            Background="Transparent">
+                      <ContentPresenter x:Name="ContentPresenter"
+                                        BorderBrush="{TemplateBinding BorderBrush}"
+                                        BorderThickness="{TemplateBinding BorderThickness}"
+                                        Content="{TemplateBinding Content}"
+                                        ContentTemplate="{TemplateBinding ContentTemplate}"
+                                        Padding="{TemplateBinding Padding}"
+                                        HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
+                                        VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}" />
+                    </Border>
+                  </ControlTemplate>
+                </Setter>
```

Comment:
Not sure if I understand why we need to override button template here. RootBorder also is not used as it seems. ContentPresenter should have all needed properties, including Background. As for me you don't need this template here, default will work.

Reply:
This part is essentially copied from upstream WinUI and also exists in the FluentAvalonia port. I'm sure you are correct, this isn't actually required (especially with the ability to drill down into the template in Avalonia). It didn't cross my mind to change it though. I can look at it if you think necessary.

That said, The lines just below where the disabled states are set are required. I can probably move that to the end with other in the Disabled State section though.

Reply:
We don't need to port controls 1to1 from WinUI with minimal changes. We need to adapt controls, so they work and looks the same (or close enough), but Avalonia-friendly.

Reply:
I know this style is used to override the Button styles that are within the template so they don't show like buttons and SplitButton can then handle the visual states itself and it's placed within the template so it doesn't bleed out into other areas. But Avalonia Selectors are more powerful so it probably would be ok to just pull that out into a separate Style and make its selector "SplitButton Button". Removing RootBorder and just setting the ContentPresenter background to Transparent is a good idea and since Selectors can target within controls

> This part is essentially copied from upstream WinUI and also exists in the FluentAvalonia port.
I tend to follow WinUI as closely as possible, as it's easier for me to diagnose if something isn't right and I can update things easier if a change is made upstream, so that's just a personal thing as to why it's in my port. Small changes like this probably aren't hard to keep up with, just never cross my mind when I'm porting.

Reply:
That a fair viewpoint. My only concern is that this is in the default style. Keeping the Fluent styles updates gets much  more difficult with these types of changes. Eventually we will need to update to Fluent v2. If Avalonia wasn't using the Fluent style, or if we were talking code-behind, I wouldn't put up much resistance here.

Reply:
I'm going to treat this comment thread as for the child Style in it's entirety. Above will only be for the disabled states which can be moved out.

Child styles like this are very common in WinUI and actually simplify quite a bit. It is useful to have this full power available to customize any way needed and also self-contined. That said, there are only three properties that are needed it seems:

```
<Setter Property="Foreground" Value="{DynamicResource SplitButtonForeground}" />
<Setter Property="BorderBrush" Value="Transparent" />
<Setter Property="BorderThickness" Value="{DynamicResource SplitButtonBorderThemeThickness}" />
```

While these could be set in other places, wouldn't we still have to set them separately for the PrimaryButton and SecondaryButton? Or would `<Style Selector="SplitButton /template/ Button>` also work here and prevent leaking these property changes to any button content developers might want (I know a Button inside a SplitButton is strange but it could happen).

-- standard disclaimer --
We are already making so many changes in the Style, I think maintaining easy compatibility with upstream Fluent / Fluentv2 changes doesn't matter now. So that decision seems to have been made -- it just requires much more work for future synchronizations.

Reply:
Thinking about this further I suppose `<Style Selector="SplitButton /template/ Button>` restricts things only to buttons within the template itself -- not to any possible Button content controls.

Reply:
Well, after testing it seems to work just fine so I made the changes. I'm not sure this is a big simplification though -- and it does complicate pulling in upstream changes (not too bad in this case).

https://github.com/AvaloniaUI/Avalonia/blob/c2cebb32fa5423ba174b9a87817f11a3c3fa40a3/src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml#L118-L146

Anyway, should be resolved.

Reply:
Please mark resolved if you agree to the changes -- I think we agree here.